sketch-on-map/src/main/java/com/esri/arcgismaps/sample/sketchonmap/MainActivity.kt

```diff
@@ -216,83 +203,75 @@ class MainActivity : AppCompatActivity() {
 
         // create a graphic from the sketch editor geometry
-        val graphic = Graphic(sketchGeometry)
-        // assign a symbol based on geometry type
-        when (sketchGeometry) {
-            is Polygon -> graphic.symbol = fillSymbol
-            is Polyline -> graphic.symbol = lineSymbol
-            is Point, is Multipoint -> graphic.symbol = pointSymbol
-            else -> {}
+        val graphic = Graphic(sketchGeometry).apply {
+            // assign a symbol based on geometry type
+            symbol = when (sketchGeometry) {
+                is Polygon -> fillSymbol
+                is Polyline -> lineSymbol
+                is Point, is Multipoint -> pointSymbol
+                else -> null
+            }
         }
 
         // add the graphic to the graphics overlay
         graphicsOverlay.graphics.add(graphic)
-        setCurrentSelectionText("Added graphic to map")
-    }
-
-    /**
-     * De-selects all buttons.
-     */
-    private fun resetButtons() {
-        activityMainBinding.pointLinePolygonToolbar.apply {
-            pointButton.isSelected = false
-            multipointButton.isSelected = false
-            polylineButton.isSelected = false
-            polygonButton.isSelected = false
-            freehandLineButton.isSelected = false
-            freehandPolygonButton.isSelected = false
-        }
     }
 
     /**
      * Clear the MapView of all the graphics and reset selections
      */
     fun clear(view: View) {
-        resetButtons()
-        geometryEditor?.clearGeometry()
-        geometryEditor?.clearSelection()
-        setCurrentSelectionText("Cleared")
+        geometryEditor.clearGeometry()
+        geometryEditor.clearSelection()
+        geometryEditor.stop()
+        selectedGeometryDropdown.setText("")
+        selectedGeometryDropdown.clearFocus()
+        showMessage(getString(R.string.cleared_message))
     }
 
     /**
-     * Called if sketch is invalid. Reports to user why the sketch was invalid.
+     * Clear all editing and applied graphics on the map
      */
-    private fun reportNotValid() {
-        val geometry = geometryEditor?.geometry?.value ?: return showError("Geometry not found")
-        val validIfText: String =
-            when (geometry) {
-                is Point -> getString(R.string.invalid_point_message)
-                is Multipoint -> getString(R.string.invalid_multipoint_message)
-                is Polyline -> getString(R.string.invalid_polyline_message)
-                is Polygon -> getString(R.string.invalid_polygon_message)
-                else -> getString(R.string.none_selected_message)
-            }
-        setCurrentSelectionText(validIfText)
+    fun restart(view: View) {
+        graphicsOverlay.graphics.clear()
+        geometryEditor.clearGeometry()
+        geometryEditor.clearSelection()
+        geometryEditor.stop()
+        selectedGeometryDropdown.setText("")
+        selectedGeometryDropdown.clearFocus()
```

Comment:
Seems like `stop` should also do this since the tool won't be selected any more. Otherwise, I find that I have to reselect the tool from the dropdown menu even though it's already selected.